core/src/main/java/org/axonframework/commandhandling/gateway/IntervalRetryScheduler.java

```diff
@@ -107,4 +125,79 @@ private boolean scheduleRetry(Runnable commandDispatch, int interval) {
             return false;
         }
     }
+
+    /**
+     * Builder class to instantiate a {@link IntervalRetryScheduler}.
+     * <p>
+     * The {@code retryInterval}, {@code maxRetryCount} and {@link ScheduledExecutorService} are
+     * <b>hard requirements</b> and as such should be provided.
+     */
+    public static class Builder {
+
+        private int retryInterval;
+        private int maxRetryCount;
+        private ScheduledExecutorService retryExecutor;
+
+        /**
+         * Sets the retry interval in milliseconds at which to schedule a retry.
+         *
+         * @param retryInterval an {@code int} specifying the retry interval in milliseconds at which to schedule a
+         *                      retry
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder retryInterval(int retryInterval) {
+            assertPositive(retryInterval, "The retryInterval must be a positive number");
+            this.retryInterval = retryInterval;
+            return this;
+        }
+
+        /**
+         * Sets the maximum number of retries allowed for a single command.
+         *
+         * @param maxRetryCount an {@code int} specifying the maximum number of retries allowed for a single command
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder maxRetryCount(int maxRetryCount) {
+            assertPositive(maxRetryCount, "The maxRetryCount must be a positive number");
+            this.maxRetryCount = maxRetryCount;
+            return this;
+        }
+
+        /**
+         * Sets the {@link ScheduledExecutorService} used to schedule a command retry.
+         *
+         * @param retryExecutor a {@link ScheduledExecutorService} used to schedule a command retry
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder retryExecutor(ScheduledExecutorService retryExecutor) {
+            assertNonNull(retryExecutor, "ScheduledExecutorService may not be null");
+            this.retryExecutor = retryExecutor;
+            return this;
+        }
+
+        /**
+         * Initializes a {@link IntervalRetryScheduler} as specified through this Builder.
+         *
+         * @return a {@link IntervalRetryScheduler} as specified through this Builder
+         */
+        public IntervalRetryScheduler build() {
+            return new IntervalRetryScheduler(this);
+        }
+
+        /**
+         * Validate whether the fields contained in this Builder are set accordingly.
```

Comment:
Should it be `Validate` or `Validates`, not sure about the form this should be in.

Reply:
Darn, you're right, should be `Validates`...I've done another 'find and replace all' to fix this.